clang-tools-extra/clangd/unittests/tweaks/ExtractFunctionTests.cpp

```diff
@@ -190,6 +190,14 @@ F (extracted();)
     }]]
   )cpp";
   EXPECT_EQ(apply(CompoundFailInput), "unavailable");
+
+  ExtraArgs.push_back("-std=c++14");
+  // FIXME: Expressions are currently not extracted
+  EXPECT_EQ(apply(R"cpp(
+                void sink(int);
+                void call() { sink([[1+1]]); }
```

Comment:
I was thinking more specifically about entire expression-statements, e.g. `[[stream << 42;]]`.

Extraction of arbitrary subexpressions would be nice as well, but likely more involved; there is a work in progress [here](https://reviews.llvm.org/D140619).

(We can keep this test case too.)

Reply:
I've added another testcase for extracting a single expression